data/abilities.js

```diff
@@ -3342,7 +3357,7 @@ exports.BattleAbilities = {
 		shortDesc: "This Pokemon cannot lose its held item due to another Pokemon's attack.",
 		onTakeItem: function (item, pokemon, source) {
 			if (this.suppressingAttackEvents() && pokemon !== this.activePokemon || !pokemon.hp || pokemon.item === 'stickybarb') return;
-			if ((source && source !== pokemon) || this.activeMove.id === 'knockoff') {
+			if ((source && source !== pokemon) || (this.activeMove && this.activeMove.id === 'knockoff')) {
```

Comment:
Maybe it would be better to ``assert(this.activeMove)`` ?

Reply:
I wouldn't mind but <code>assert()</code> currently isn't used anywhere outside of the test suite. @Zarel which would you prefer?

Reply:
PS actually does assert all over the codebase, it's just that we don't use 

    assert(shouldBeTrue);

we use

    if (!shouldBeTrue) throw new Error("shouldBeTrue isn't true");

The latter _is_ used all over the PS codebase, and is probably the best choice here, assuming you actually want to assert here.